.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,33 +1,70 @@
 Hi! Before submitting your PR for review, and/or before merging it, please
-go through the following checklist:
+go through the following checklists:
 
-- [ ] For DB changes, check for missing indexes, check to see if the changes
-affect other apps (such as the dashboard), make sure the DB columns in the
-various environments are properly populated, coordinate with devops, plan
-migrations in separate steps.
+### General
 
-- [ ] For route changes, make sure GET requests don't change state or result in
-destructive behavior. GET requests should only result in information being
-read, not written.
+- [ ] `make lint` ran locally without errors
 
-- [ ] For encryption changes, make sure it is compatible with data that was
-encrypted with the old code.
+- [ ] Rubocop or Reek offenses were not disabled unless they are false positives.
+If you're not sure, please ask a teammate.
 
-- [ ] For secrets changes, [make sure to update the S3 secrets bucket](https://github.com/18F/identity-private/wiki/Secrets-S3-buckets) with the 
-new configs in **all** environments. 
+- [ ] Tests added for this feature/bug
 
-- [ ] Do not disable Rubocop or Reek offenses unless you are absolutely sure
-they are false positives. If you're not sure how to fix the offense, please
-ask a teammate.
+- [ ] The PR adheres to our [PR guidelines]
+
+- [ ] Related commits have been squashed
+
+- [ ] Method names in classes are in descending order of abstraction.
+See https://youtu.be/0rsilnpU1DU?t=554
+
+### Database
+
+- [ ] Indexes were added if necessary
+
+- [ ] Verified that the changes don't affect other apps (such as the dashboard)
+
+- [ ] Changes were made to allow the migration to happen over several deploys to
+avoid downtime and migration errors
+
+- [ ] An issue has been opened as a reminder to populate the necessary DB columns
+in the various environments right before deploying, taking into account the users
+who might not have interacted with this column yet (such as users who have not
+set a password yet)
+
+- [ ] The migrations have been timed and tested against a copy of the production
```

Comment:
This strikes me as a bit heavy handed. Migrations shouldn't take long enough to run to require this much ceremony. Things that are applying transformations to the data that will take a while to run may be better suited to an out of band database chore.

Reply:
If you recall, we had a migration a while back that was interrupted due to the 2.5s statement timeout we had set, and which caused issues in production while deploying RC 56: https://github.com/18F/identity-idp/commit/17130437760d155b3385c30c7fe7048d0cebaf90

Reply:
Also, these are not meant to be mandatory on every PR. A migration that drops columns probably doesn't need to be timed. These are things to keep in mind when working on and reviewing a PR.

Reply:
:arrow_up: Maybe add the above qualification to the guideline...

Reply:
Actually, I can't say that with confidence. The last migration that caused errors did something that seemed trivial: adding a new column and index to the Users table. I think it's better to be safe than sorry. I will update the text to remove references to the timing as the important thing is to make sure this won't blow up when we deploy.